scripts/validate-version.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+# Write output to error output stream.
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
+die()
+{
+  local _return="${2:-1}"
+  echo_stderr "$1"
+  exit "${_return}"
+}
+
+set -euo pipefail
+
+# Set the path to the Chart.yaml file
+SCRIPT_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]:-"$0"}")")"
+ROOT_DIR="$SCRIPT_DIR/.."
+CHART_DIR="$ROOT_DIR/deploy/helm/charts"
+CHART_YAML="$CHART_DIR/Chart.yaml"
+
+# Check if the Chart.yaml file exists
+if [ ! -f "$CHART_YAML" ]; then
+  die "Chart.yaml file not found in $CHART_YAML"
+fi
+
+# Extract the chart version and app version using yq
+CHART_VERSION=$(yq e '.version' "$CHART_YAML")
+APP_VERSION=$(yq e '.appVersion' "$CHART_YAML")
+
+# Check if extraction was successful
+if [ -z "$CHART_VERSION" ] || [ -z "$APP_VERSION" ]; then
+  die "Failed to extract versions from Chart.yaml"
+fi
+
+# Print the extracted versions
+echo "Chart Version: $CHART_VERSION"
+echo "App Version: $APP_VERSION"
+
+# Validate that the versions are valid semver
+if [ "$(semver validate "$CHART_VERSION")" != "valid" ]; then
+  die "Invalid chart version: $CHART_VERSION"
+fi
+
+if [ "$(semver validate "$APP_VERSION")" != "valid" ]; then
+  die "Invalid app version: $APP_VERSION"
+fi
+
+help() {
+  cat <<EOF
+Usage: $(basename "$0") [OPTIONS]
+
+Options:
+  --branch <branch name>                    Name of the branch on which this workflow is running.
+
+Examples:
+  $(basename "$0") --branch develop
+EOF
+}
+
+die() {
```

Comment:
Duplicate of the die function up above?

Reply:
Ohh, I initially used that but then used die and forgot to remove it.